### PR TITLE
feat: Wipe wallet group

### PR DESCRIPTION
**Pull request by BrodyHughes**
Fixes BX-1254
Figma link: https://www.figma.com/file/hzfwKOddfRmeVO4KUsG9wM/branch/bQjlkhAqO2w3nk7NJF5EKS/Rainbow-BX?type=design&node-id=15871-3647&mode=design&t=XOaSEV6pKSICFPvc-0

## What changed (plus any additional context for devs):
- added the ability to remove an entire wallet group via wallet details
- should only show up for HD wallets atm
- watched are uneffected
- will follow up with the ability to unpair a ledger group next

## Screen recordings / screenshots

POW:
<video src="https://github.com/rainbow-me/browser-extension/assets/41711440/57ee0fdd-1f15-4dc4-8e04-0e98d4112301" />

### CODE REVIEW

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
UI looks great! Noticed an issue where the Wallet Group would re-appear in the Wallets & Keys menu initially after completing the flow. The group would shift from "Backed Up" to the "Not backed up" state. Then when completing the flow again, it would be fully removed
Screenshot 2024-02-05 at 12 59 25 AM

**Comment by BrodyHughes** [comment]
> UI looks great! Noticed an issue where the Wallet Group would re-appear in the Wallets & Keys menu initially after completing the flow. The group would shift from "Backed Up" to the "Not backed up" state. Then when completing the flow again, it would be fully removed Screenshot 2024-02-05 at 12 59 25 AM

cannot repro.

**Comment by DanielSinclair** [comment]
@BrodyHughes Found another issue where dapps connections persist after we remove a seed phrase, leading to these keychain errors. This might be true for "hide wallet" and "remove wallet" features too. I think we'll need to implement a watcher or logic to automatically disconnect dapps in these scenarios Screenshot 2024-02-06 at 9 47 02 PM

File: src/entries/popup/components/WarningInfo/walletGroupWipeWarningInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+interface WarningInfoProps {
+  iconAndCopyList: IconAndCopyItem[];
+  onProceed: () => void;
+  testId?: string;
+}
+
+export default function WarningInfo({
+  iconAndCopyList,
+  onProceed,
+  testId,
```
Comment (greg-schrammel) [inline src/entries/popup/components/WarningInfo/walletGroupWipeWarningInfo.tsx:37]: if you don't have another use case for a different testId I don't think it needs to be a prop

File: src/entries/popup/components/WarningInfo/walletGroupWipeWarningInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  i18n.t(s, { scope: 'settings.privacy_and_security.wallets_and_keys' });
+
+interface WarningInfoProps {
+  iconAndCopyList: IconAndCopyItem[];
+  onProceed: () => void;
+  testId?: string;
+}
+
+export default function WarningInfo({
+  iconAndCopyList,
```
Comment (greg-schrammel) [inline src/entries/popup/components/WarningInfo/walletGroupWipeWarningInfo.tsx:35]: maybe this don't have to be a prop either, seems like a very specific component no need to abstract it, or change the name and abstract to use in the other similar screens?

**Review by greg-schrammel — APPROVED** [review]
🐐🐐🐐

File: src/entries/popup/pages/settings/walletsAndKeys/walletWipePrompt.tsx [hunk 3]
```diff
@@ -22,25 +18,18 @@ import * as wallet from '../../../handlers/wallet';
 const t = (s: string) =>
   i18n.t(s, { scope: 'settings.privacy_and_security.wallets_and_keys' });
 
+async function handleWipeWallet() {
```
Comment (BrodyHughes) [inline src/entries/popup/pages/settings/walletsAndKeys/walletWipePrompt.tsx:21]: could maybe inline this now bc it's way smaller now with your refactor. idk what your preference is.

Comment (greg-schrammel) [reply src/entries/popup/pages/settings/walletsAndKeys/walletWipePrompt.tsx:21]: I like creating functions outside the component when possible, so it won't be recreated on every rerender, but it's not a huge performance gain here I'd go with whatever you think is more readable, up to you

**Review by BrodyHughes — COMMENTED** [review]
tested this with greg's changes and it lgtm

**Review by DanielSinclair — APPROVED** [review]
Looks good to me!

**Pull request merged**